### ISSUE TITLE
ag-Grid: Framework component is missing the method refresh()

### ISSUE THREAD

**Issue by maccuaa** [ag-grid/ag-grid-react#95]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid-react/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Calling [refreshCells](https://www.ag-grid.com/javascript-grid-refresh/#gsc.tab=0) or [updateRowData](https://www.ag-grid.com/javascript-grid-data-update/#gsc.tab=0) causes the following warning to be displayed: *ag-Grid: Framework component is missing the method refresh()*

**Expected behavior**

No warning should be displayed for calling the APIs as documented in the documentation.

**Minimal reproduction of the problem with instructions**

[http://plnkr.co/edit/OwSC4uqUUuyOQI9rrzL1](http://plnkr.co/edit/OwSC4uqUUuyOQI9rrzL1)

* **ag-grid-react version:** 13.1.0

* **ag-Grid/ag-Grid-Enterprise version:** 13.1.2

* **React version:** 15.6.1

**Comment by seanlandsman** [comment]
Hi,

Thanks for raising this. I've raised AG-823 our side to track this - you can view its progress on our pipeline page here: https://www.ag-grid.com/ag-grid-pipeline/

We'll update this thread with updates as soon as we have any.

thanks

**Comment by jintoppy** [comment]
I don't see the bug number AG-823 in the pipeline page. Can you please check

**Comment by seanlandsman** [comment]
@jintoppy Hi - we have an issue where jira is limiting outout to 100 issues, which is why you're not seeing this issue

However, I've taken a look at this issue and the message is correct - the refresh method is now mandatory.  That said the message and docs should be updated to make this clearer.  I've modified the ticket to reflect this

tl;dr - the refresh method is now mandatory in renderer components

**Comment by maccuaa** [comment]
@seanlandsman  - Thanks for looking into this, yes the docs definitely need to be updated then since they currently say:

> You have the option of handling refresh or not by either providing a refresh() method on your React component or not. If not present, then the grid will destroy your component and create a new one if it tries to refresh the cell.

**Comment by ymjongbin** [comment]
I found a solution to avoid the issue which is "ag-Grid: Framework component is missing the method refresh()".

Add this code in your editor or viewer component
I am using "ag-grid" and "ag-grid-angular" version 13.0.x

// In YourViewerComponent
    public refresh(): void {
        // empty is OK
    }


// Column definition
    cellEditorFramework: YourViewerComponent,
    cellRendererFramework: YourViewerComponent

// Called from your root component for calling a refreshcell 
const count = this.gridOptions.api.getDisplayedRowCount();
for (let i = 0; i < count; i++) {
    rowNode = this.gridOptions.api.getDisplayedRowAtIndex(i);
    const rownodes = [rowNode]; // params needs an array
    const params = {
        force: true,
        columns: ['yourCoulunFieldName'],
        rowNodes: rownodes
    };
    console.log('refresh ');
    this.gridOptions.api.refreshCells(params);
}

**Comment by ceolter** [comment]
i've updated the docs (will be in next release). refresh method is mandatory for the frameworks. this was introduced in the version of ag-Grid where we updated to a newer type script version, where typescript no longer likes interfaces with only optional methods. to support the latest type script, we were forced to make this method mandatory. the fix, as above, is to have a refresh method in your component. if you don't want to handle the refresh, then you should return 'false' (if you return true, or return nothing, then the grid thinks you handled the refresh).

**Comment by hpeikari** [comment]
This issue wasted my whole day ! This seems like a smelly code/library to me !

**Comment by jjarcik** [comment]
Hi, 
this issue is same in ag-grid 15.0.0. I have this component in React 

```JS
class GridCell extends Component {
  refresh(params) {
    return true
  }
  render () {
    return <span>{this.props.value}</span>
  }
}

export default GridCell
```

and still getting the warning **ag-Grid: Framework component is missing the method refresh()**

Is there something wrong in my code?

**Comment by adambisek** [comment]
@jjarcik 
This is sufficient.
You have probably your GridCell component wrapped with some HoC, that makes from your component a function (and therefore there is not available refresh method).
